designcompose/src/main/java/com/android/designcompose/DesignView.kt

```diff
@@ -309,19 +315,25 @@ internal fun DesignView(
     val onPressReactions: MutableList<Reaction> = ArrayList()
     val onDragReactions: MutableList<Reaction> = ArrayList()
     val onKeyReactions: MutableList<Reaction> = ArrayList()
+
     view.reactions.ifPresent { reactions ->
         for (reaction in reactions) {
-            when (reaction.trigger) {
-                is Trigger.OnClick -> onClickReactions.add(reaction)
-                is Trigger.OnPress -> onPressReactions.add(reaction)
-                is Trigger.OnDrag -> onDragReactions.add(reaction)
-                is Trigger.AfterTimeout -> {
-                    if ((reaction.trigger as Trigger.AfterTimeout).timeout < currentTimeout) {
-                        onTimeout = reaction
-                        currentTimeout = (reaction.trigger as Trigger.AfterTimeout).timeout
+            reaction.trigger.type?.run {
+                when (this) {
+                    is TriggerType.Click -> onClickReactions.add(reaction)
+                    is TriggerType.Press -> onPressReactions.add(reaction)
+                    is TriggerType.Drag -> onDragReactions.add(reaction)
+                    is TriggerType.AfterTimeout -> {
+                        if (value.timeout < currentTimeout) {
+                            onTimeout = reaction
+                            currentTimeout = value.timeout
+                        } else {
```

Comment:
We don't really need this `else {{}}` right? Remove?

Reply:
The extra {} isn't needed, but the else is needed when used in a `when`, apparently.